### ISSUE TITLE
Feature request: svg support

### ISSUE THREAD

**Issue by jplecavalier** [wilkelab/gridtext#15]
Hi, thanks for the great job!

I'm wondering if it would be possible for you to add support for svg image within the internal `read_image()` function. For now, only png and jpeg are supported.

In the end, I'd like to be able to add svg logos as axis label in a ggplot using `element_markdown()` from [`ggtext`](https://wilkelab.org/ggtext/). Let me know if you need a reprex, but I'm sure you get it without one.

Thanks again!

**Comment by clauswilke** [comment]
That's basically impossible given the current state of R graphics devices. If R graphics devices start supporting svg images then I can hand an image through.

**Comment by jplecavalier** [comment]
I totally believe you since I have a very limited understanding of what's going on under the hood of graphic devices. I just want to point to [this SO answer](https://stackoverflow.com/a/61466576/5187123) if this could help by any chance.

**Comment by clauswilke** [comment]
Yeah, I'm aware of grImport2. I feel it's too much of a compromise to be worth supporting. The moment people try to use it to import a real-world svg they've come across it'll not render correctly. I'd want to be able to at least pass svg's through to svg-aware graphics devices, such as svg. On the flip side, if your output device is a bitmap device, then pre-rendering the svg as a bitmap is fine also. So the subset of cases where you need vector graphics and are Ok with a very limited subset of the svg standard is rather small.

**Comment by clauswilke** [comment]
In any case, development on this code base is basically frozen, because there are many other features I'd like to support that can't be supported for various reasons.

The next iteration exists in a draft version, but it'll probably be 2-3 years before this is ready for prime time: https://clauswilke.com/sinab/
